packages/dashboard/src/components/dashboard/main-menu.tsx

```diff
@@ -16,37 +16,44 @@ const useStyles = makeStyles((theme) => ({
 
 export interface MainMenuProps {
   pushView(view: OmniPanelViewIndex): void;
+  setSearch?: () => void;
```

Comment:
I think we should have some docs for this prop.

On another note, this brings up some design consideration I have:

Should `MainMenu` manage the search state? Would it be better to expose on click handlers for each button and let `Dashboard` handle it? This also goes for `pushView`. I feel doing so is more "react-ive". The counter argument is that since there is no plan to use `MainMenu` outside of `Dashboard`, the usual react best practices does not offer any advantage, and doing it this way reduces the load/bloat on `Dashboard` and avoid it becoming somewhat of a "god class".

reference: https://reactjs.org/docs/thinking-in-react.html#step-4-identify-where-your-state-should-live

@matiasbavera Which do you think is the better approach?